File: rvbsim.dnp/Helpers/StringCallback.cs

using System.Threading.Tasks;
using Automatak.DNP3.Interface;

namespace rvb_sim.dnp
{
    /// <summary>
    ///
    /// </summary>
    public class StringCallback : ITaskCallback

    {
        readonly TaskCompletionSource<TaskCompletion> tcs = new TaskCompletionSource<TaskCompletion>();

        public Task<TaskCompletion> Task => tcs.Task;

        #region Constructor

        /// <summary>
        /// Default constructor
        /// </summary>
        public StringCallback()
        {

        }

        public void OnComplete(TaskCompletion result)
        {
            tcs.SetResult(result);
        }

        public void OnDestroyed()
        {
            //throw new System.NotImplementedException();
        }

        public void OnStart()
        {
           // throw new System.NotImplementedException();
        }

        #endregion
    }
}

File: rvbsim/dnp3/Helpers/StringCallback.vb
Imports System.Threading.Tasks
Imports Automatak.DNP3.Interface

Public Class StringCallback
    Implements ITaskCallback

    ReadOnly tcs As TaskCompletionSource(Of TaskCompletion) = New TaskCompletionSource(Of TaskCompletion)

    Public ReadOnly Property Task As Task(Of TaskCompletion)

    Public Sub OnStart() Implements ITaskCallback.OnStart
        'Do nothing extra at OnStart()
    End Sub

    Public Sub OnComplete(result As TaskCompletion) Implements ITaskCallback.OnComplete
        tcs.SetResult(result)
    End Sub

    Public Sub OnDestroyed() Implements ITaskCallback.OnDestroyed
        'Do nothing extra at OnStart()
    End Sub
End Class
